pydantic/env_settings.py

```diff
@@ -27,20 +29,66 @@ def __init__(
         __pydantic_self__,
         _env_file: Union[Path, str, None] = env_file_sentinel,
         _env_file_encoding: Optional[str] = None,
+        _secrets_dir: Union[Path, str, None] = None,
         **values: Any,
     ) -> None:
         # Uses something other than `self` the first arg to allow "self" as a settable attribute
         super().__init__(
-            **__pydantic_self__._build_values(values, _env_file=_env_file, _env_file_encoding=_env_file_encoding)
+            **__pydantic_self__._build_values(
+                values, _env_file=_env_file, _env_file_encoding=_env_file_encoding, _secrets_dir=_secrets_dir
+            )
         )
 
     def _build_values(
         self,
         init_kwargs: Dict[str, Any],
         _env_file: Union[Path, str, None] = None,
         _env_file_encoding: Optional[str] = None,
+        _secrets_dir: Union[Path, str, None] = None,
     ) -> Dict[str, Any]:
-        return deep_update(self._build_environ(_env_file, _env_file_encoding), init_kwargs)
+        return deep_update(
+            self._build_secrets(_secrets_dir), self._build_environ(_env_file, _env_file_encoding), init_kwargs
+        )
+
+    def _build_secrets(self, _secrets_dir: Union[Path, str, None] = None) -> Dict[str, Optional[str]]:
+        """
+        Build secrets suitable for passing to the Model.
+        """
+        d: Dict[str, Optional[str]] = {}
+
+        secrets_dir = _secrets_dir or self.__config__.secrets_dir
+        if secrets_dir is None:
+            return d
+
+        secrets_path = Path(secrets_dir)
+
+        if not secrets_path.is_dir():
```

Comment:
surely to avoid a lot of confusion we should raise an error if `secrets_path.is_file()`?

Reply:
perhaps we should also raise an error/warning if the directory is missing? i'm not so sure on that.

Perhaps only if it's provided via an argument, not config?

Reply:
Good point, I think that since the default value for secrets_dir is None, that providing any value pointing to a directory that is missing can be an error.